lib/src/node/compile.dart

```diff
@@ -260,18 +263,24 @@ List<AsyncCallable> _parseFunctions(Object? functions, {bool asynch = false}) {
     if (!asynch) {
       late Callable callable;
       callable = Callable.fromSignature(signature, (arguments) {
-        var result =
-            simplify((callback as Function)(toJSArray(arguments)) as Object);
-        if (result is Value) return result;
+        var result = (callback as Function)(toJSArray(arguments));
         if (isPromise(result)) {
           throw 'Invalid return value for custom function '
               '"${callable.name}":\n'
               'Promises may only be returned for sass.compileAsync() and '
               'sass.compileStringAsync().';
-        } else {
-          throw 'Invalid return value for custom function '
-              '"${callable.name}": $result is not a sass.Value.';
         }
+        if (result is Value) {
+          var simplified = simplify(result);
+          if (simplified is! Value) {
+            throw 'Custom function "${callable.name}" '
+                'returned an object that cannot be simplified to a '
+                'sass.Value.';
```

Comment:
As it stands, I think this throw is unreachable. If you call `simplify(Value)` it will always return a `Value`. This suggests to me that we should actually have two different functions:

```dart
Value _simplifyValue(Value value);
Object _simplifyCalcArg(Object object);
```

which call each other recursively as necessary.